pinot-core/src/main/java/org/apache/pinot/core/common/ObjectSerDeUtils.java

```diff
@@ -474,21 +477,31 @@ public byte[] serialize(HyperLogLog hyperLogLog) {
     @Override
     public HyperLogLog deserialize(byte[] bytes) {
       try {
-        return HyperLogLog.Builder.build(bytes);
-      } catch (IOException e) {
-        throw new RuntimeException("Caught exception while de-serializing HyperLogLog", e);
+        IntBuffer intBuf =
+                ByteBuffer.wrap(bytes)
+                        .order(ByteOrder.BIG_ENDIAN)
+                        .asIntBuffer();
+        // The first 2 bytes are constant headers for the HLL. We only need the first byte, the log2m.
```

Comment:
(minor) The first 2 integers are ... need the first integer...

Reply:
Fixed